src/mesh/Data.hpp

```diff
@@ -83,7 +100,13 @@ class Data {
   /// Sets all values to zero
   void toZero();
 
-  /// Returns the dimension (i.e., number of components) of one data value.
+  /// Returns if the data contains gradient data
+  bool hasGradient() const;
+
+  /// Returns the mesh dimension (i.e., number of rows) of one gradient data value .
```

Comment:
Isn't the mesh dimension the same as the overall spatial dimension? Maybe the words 'spatial' and 'data' help to distinguish here.